### PR TITLE
Security second attempt

### PR DESCRIPTION
**Pull request by Garoma22**
Implemented:

JWT Authentication: JWT token generation and validation were set up, allowing secure communication between the client and server.

Password Encryption: Password encryption during user registration was implemented, and the authentication process works with encrypted passwords.

Custom Trainee Registration: Functionality was added to register a Trainee with automatic username and password generation, properly storing encrypted passwords in the database.

Security Configuration: Spring Security was configured to enable stateless sessions, permit access to certain endpoints (e.g., /auth/**) without authentication, and secure others.

Brute force protector added

### CODE REVIEW

File: src/main/java/com/example/gymApp/config/CorsConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  public CorsFilter corsFilter() {
+    CorsConfiguration config = new CorsConfiguration();
+    config.setAllowCredentials(true);
+    config.setAllowedOrigins(Arrays.asList(
+        "http://localhost:8080",
+        "http://localhost:8081",
+        "http://localhost:8082",
+        "http://localhost:8083",
+        "http://localhost:8084"));
+    config.setAllowedMethods(Arrays.asList("GET", "POST", "PUT", "DELETE", "OPTIONS", "PATCH", "HEAD"));
```
Comment (HannaVatlina) [inline src/main/java/com/example/gymApp/config/CorsConfig.java]: Use List.of() for creation a list

File: src/main/java/com/example/gymApp/config/CorsConfig.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    config.setAllowedHeaders(Arrays.asList("Authorization", "Cache-Control", "Content-Type"));
```
Comment (HannaVatlina) [inline src/main/java/com/example/gymApp/config/CorsConfig.java]: Use constants from the HttpHeaders class

File: src/main/java/com/example/gymApp/config/SecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  private final JwtAuthenticationFilter jwtAuthFilter;
+  private final AuthenticationProvider authenticationProvider;
+
+  @Bean
+  SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http
+        .csrf(AbstractHttpConfigurer::disable)
+        .authorizeHttpRequests(authorize -> authorize
+            .requestMatchers("/auth/**").permitAll()
+            .anyRequest().authenticated()
```
Comment (HannaVatlina) [inline src/main/java/com/example/gymApp/config/SecurityConfiguration.java:31]: all your requests under authorization even if I want to see the list of training types to create a trainer

File: src/main/java/com/example/gymApp/model/User.java [hunk 4]
```diff
@@ -42,5 +54,32 @@ public User(Long id, String firstName, String lastName, String username, String
...
 
+  @Override
+  public boolean isCredentialsNonExpired() {
+    return true;
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
```
Comment (HannaVatlina) [inline src/main/java/com/example/gymApp/model/User.java]: why you need all those methods?

File: src/main/java/com/example/gymApp/config/SecurityConfiguration.java (hunk 3 above)
Comment (Garoma22) [reply src/main/java/com/example/gymApp/config/SecurityConfiguration.java:31]: done

File: src/main/java/com/example/gymApp/config/CorsConfig.java (hunk 2 above)
Comment (Garoma22) [reply src/main/java/com/example/gymApp/config/CorsConfig.java]: done

**Pull request merged**